From 5283cfcd44c38dd7133fa829f1c09a584d288bd5 Mon Sep 17 00:00:00 2001
From: Yongbo Zhang <yongbo.zhang.xh@renesas.com>
Date: Tue, 10 May 2022 10:33:08 +0800
Subject: [PATCH 29/34] soc: renesas: rcar-sysc: always on A2VC power domain

In cr7-solution, cr7 boots before Linux/Android and enables the A2VC power
domain to ensure that the decoder works.

Unfortunately, the default configuration of Linux/Android is to turn off the
A2VC power domain, which is fatal for the cr7 decoder to work.

So this patch forces the A2VC power domain on to ensure the cr7 decoder keeps
working properly.
---
 drivers/soc/renesas/rcar-sysc.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/drivers/soc/renesas/rcar-sysc.c b/drivers/soc/renesas/rcar-sysc.c
index 3c6dc1edf01e..4cf81de30ec9 100644
--- a/drivers/soc/renesas/rcar-sysc.c
+++ b/drivers/soc/renesas/rcar-sysc.c
@@ -65,6 +65,15 @@
 
 static
 const struct soc_device_attribute rcar_sysc_quirks_match[] __initconst = {
+	{
+		.soc_id = "r8a7795", .revision = "ES3.*",
+		.data = (void *)(BIT(R8A7795_PD_A3VP) | BIT(R8A7795_PD_CR7)
+			| BIT(R8A7795_PD_A3VC) | BIT(R8A7795_PD_A2VC0)
+			| BIT(R8A7795_PD_A2VC1) | BIT(R8A7795_PD_A3IR)
+			| BIT(R8A7795_PD_3DG_A) | BIT(R8A7795_PD_3DG_B)
+			| BIT(R8A7795_PD_3DG_C) | BIT(R8A7795_PD_3DG_D)
+			| BIT(R8A7795_PD_3DG_E)),
+	},
 	{
 		.soc_id = "r8a7795", .revision = "ES2.0",
 		.data = (void *)(BIT(R8A7795_PD_A3VP) | BIT(R8A7795_PD_CR7)
@@ -84,6 +93,13 @@ const struct soc_device_attribute rcar_sysc_quirks_match[] __initconst = {
 			| BIT(R8A7795_PD_3DG_E)),
 
 	},
+	{
+		.soc_id = "r8a7796", .revision = "ES3.*",
+		.data = (void *)(BIT(R8A7796_PD_CR7) | BIT(R8A7796_PD_A3VC)
+			| BIT(R8A7796_PD_A2VC0) | BIT(R8A7796_PD_A2VC1)
+			| BIT(R8A7796_PD_A3IR) | BIT(R8A7796_PD_3DG_A)
+			| BIT(R8A7796_PD_3DG_B)),
+	},
 	{
 		.soc_id = "r8a7796", .revision = "ES1.*",
 		.data = (void *)(BIT(R8A7796_PD_CR7) | BIT(R8A7796_PD_A3VC)
@@ -91,6 +107,13 @@ const struct soc_device_attribute rcar_sysc_quirks_match[] __initconst = {
 			| BIT(R8A7796_PD_A3IR) | BIT(R8A7796_PD_3DG_A)
 			| BIT(R8A7796_PD_3DG_B)),
 	},
+	{
+		.soc_id = "r8a77965", .revision = "ES1.*",
+		.data = (void *)(BIT(R8A7796_PD_CR7) | BIT(R8A7796_PD_A3VC)
+			| BIT(R8A77965_PD_A3VP) | BIT(R8A77965_PD_A2VC1)
+			| BIT(R8A77965_PD_3DG_A) | BIT(R8A77965_PD_3DG_B)),
+	},
+
 	{ /* sentinel */ }
 };
 
-- 
2.17.1

